### DISCUSSION TITLE
How to block indexing of custom preview domain?

### DISCUSSION THREAD

**Discussion by OneX** [vercel/community#591] [Help] [answered]
Hey, can anybody suggest how to block indexing of custom domain i made for preview of the project?

Thanks!

**Comment by swarnava** [comment, marked as answer]
If you are using Next.js, in `next.config.js` file, you can try adding following config:

```js
webpack: (config, options) => {
    // Generate disallow all robots.txt and add to /public folder in preview
    if (options.isServer && process.env.VERCEL_ENV === 'preview') {
      require('./scripts/generate-robotstxt-preview.js');
    }

//other webpack config here
}
```

This adds the generated `robots.txt` file which disallows all crawlers (the ones that actually follow the robots directions) from crawling the preview sites, custom domain or not.
